src/regressionmodel.jl

```diff
@@ -65,11 +71,13 @@ of each data point.
 function cooksdistance end
 
 """
-    residuals(model::RegressionModel)
+    residuals(model::RegressionModel; weighted::Bool=false)
+
+Return the residuals of the model or, if `weighted=true`, the residuals multiplied by
+the square root of the [model weights](@ref weights(::StatisticalModel)). 
```

Comment:
Where does the square root come from exactly? Doesn't that assume a particular definition of residuals (i.e. using L2-norm rather than e.g. L1-norm)?

Reply:
Well, this is tricky. Also `modelmatrix` multiplies the entries of `X` by the square-root of the weights. Why?

Think about the linear model. With weights, the crossmodel matrix is $X'WX$. Then, to obtain it we can now do `modelmatrix(lm1; weighted = true)'modelmatrix(lm1; weighted = true)`. 

Notice that this is consistent with `R`; see, e.g., the function [weighted.residuals](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/weighted.residuals.html) which is in stats.

With weights, any weights is 
$$\sqrt{w_i}y_i = \sqrt{w_i}x_i \beta + \sqrt{w_i}u_i.$$ So the understanding is that weighting single constituents of the model (y,x,u) amount to weight by $\sqrt{w_i}$.

Reply:
Yeah, another tricky point. My understanding is that for residuals, the square root comes from the fact that deviance residuals themselves are defined as the square root of quantities which are partitions of the deviance. Right?

Note the R docstring for `weighted.residuals` says "Weighted residuals are based on the deviance residuals", which are only one kind of residual. Actually in R `residuals` also returns weighted residuals, except for response residuals, which are always unweighted. Maybe to be completely accurate we could say "for deviance and Pearson residuals...", so that packages are free to use different definitions (or throw an error) if needed?

Reply:
I think what @nalimilan says is that the assumption here (and in your change of `modelmatrix`) is that for all kinds of weights the weighted model matrix is `X * sqrt.(W)`. Is it always true for `FrequencyWeights`, `AnalyticWeights` and `ProbabilityWeights`? x-ref: https://github.com/JuliaStats/GLM.jl/pull/487

Reply:
@bkamins weighted residuals, weighted model matrix do not exist in statistics. They are only useful from a coding point of view - they make it easier to write neater code. 

I have always defined these quantities as multiplied by $\sqrt{w_i}$ as it is much more convenient. Some thing for R — which returns silently squared-root weighted residuals. Also other packages, notable FixedEffectModels.jl does that. 

@nalimilan make sense what you propose - I will add more context to the doc

Reply:
Even if these don't exist in statistics, the question can be phrased as "are there situations where the returned value is useful, even when you don't know the kind of weights used". I think the answer is yes, but it's tricky, so... R base only supports analytic weights so it's not a great reference.

Reply:
OK let's just adapt the docstring then. Feel free to add more context if you have ideas.